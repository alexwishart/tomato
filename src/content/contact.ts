export const contactContent = {
  title: 'Keep up to date on the tomato',
  subtitle:
    'Enter your name and email address below to receive weekly updates on the status of the tomato',
  inputFields: [
    { value: 'name', label: 'Name' },
    { value: 'email', label: 'E-mail' },
  ],
  submitButtonLabel: 'Submit',
};
