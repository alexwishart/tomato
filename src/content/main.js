import tomatoImage from '../images/tomato.jpg';

export const mainContent = {
  title: 'The Tomato',
  image: tomatoImage,
};
