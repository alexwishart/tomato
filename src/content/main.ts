import tomatoImage from '../images/tomato.jpg';

export const mainContent = {
  title: 'The Tomato',
  image: { src: tomatoImage, alt: 'Image of tomato' },
  content: {
    title: 'Varieties',
    paragraphs: [
      "There are around 7500 tomato varieties grown for various purposes. Heirloom tomatoes are becoming increasingly popular, particularly among home gardeners and organic producers, since they tend to produce more interesting and flavorful crops at the cost of disease resistance and productivity. In 1973, Israeli scientists developed the world's first long shelf-life commercial tomato varieties.",
      'Tomato varieties are roughly divided into several categories, based mostly on shape and size.',
    ],
    varieties: [
      '"Slicing"/"globe" tomatoes',
      'Beefsteak tomatoes',
      'Oxheart tomatoes',
      'Plum tomatoes',
      'Pear tomatoes',
      'Cherry tomatoes',
      'Grape tomatoes',
      'Campari tomatoes',
    ],
  },
};
