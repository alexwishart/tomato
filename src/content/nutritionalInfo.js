export const nutritionalInfoContent = {
  title: 'Red tomatoes, raw',
  subtitle: 'Nutritional value per 100 g (3.5 oz)',
  table: {
    header: ['Energy', '74kJ (18 kcal)'],
    rows: [
      ['Carbohydrates', '3.9 g'],
      ['- Sugars', '2.6 g'],
      ['- Dietary fiber', '1.2 g'],
      ['Fat', '0.2 g'],
      ['Protein', '0.9 g'],
      ['Water', '94.5 g'],
      ['Vitamin A equiv.', '42 µg (5%)'],
      ['- lutein and zeaxanthin', '123 µg'],
      ['Vitamin C', '14 mg (17%)'],
      ['Vitamin E', '0.54 mg (4%)'],
      ['Potassium', '237 mg (5%)'],
    ],
  },
  footer: {
    paragraphs: [
      'Percentages are relative to US recommendations for adults',
      'Source:',
    ],
    link: {
      text: 'USDA Nutrient Database',
      href: 'http://www.nal.usda.gov/fnic/foodcomp/search/',
    },
  },
};
