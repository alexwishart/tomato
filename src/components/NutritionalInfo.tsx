import React from 'react';
import styled from 'styled-components';
import { nutritionalInfoContent } from '../content/nutritionalInfo';

const NutritionalInfo = () => {
  return (
    <Container>
      <h3>{nutritionalInfoContent.title}</h3>
      <p>{nutritionalInfoContent.subtitle}</p>
      <table>
        <tbody>
          <tr>
            <TableHeader>{nutritionalInfoContent.table.header[0]}</TableHeader>
            <TableHeader>{nutritionalInfoContent.table.header[1]}</TableHeader>
          </tr>
          {nutritionalInfoContent.table.rows.map((row) => {
            return (
              <tr key={row[0]}>
                <Cell>{row[0]}</Cell>
                <Cell>{row[1]}</Cell>
              </tr>
            );
          })}
        </tbody>
      </table>
      <p>{nutritionalInfoContent.footer.paragraphs[0]}</p>
      <p>
        {nutritionalInfoContent.footer.paragraphs[1]}{' '}
        <a href={nutritionalInfoContent.footer.link.href}>
          {' '}
          {nutritionalInfoContent.footer.link.text}
        </a>
      </p>
    </Container>
  );
};

const Container = styled.div`
  width: 30%;
`;

const TableHeader = styled.th`
  background: #ebebeb;
`;

const Cell = styled.td`
  padding-top: 1rem;
`;

export default NutritionalInfo;
