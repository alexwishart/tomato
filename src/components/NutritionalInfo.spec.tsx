import React from 'react';
import { render } from '@testing-library/react';
import NutritionalInfo from './NutritionalInfo';
import { nutritionalInfoContent } from '../content/nutritionalInfo';

describe('The NutritionalInfo component', () => {
  it('should render correctly', () => {
    expect(render(<NutritionalInfo />)).toMatchSnapshot();
  });

  it('should redirect to the usda website when the footer link is clicked', () => {
    const { getByText } = render(<NutritionalInfo />);

    expect(getByText(nutritionalInfoContent.footer.link.text).href).toBe(
      nutritionalInfoContent.footer.link.href
    );
  });
});
