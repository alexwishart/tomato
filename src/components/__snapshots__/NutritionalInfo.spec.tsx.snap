// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NutritionalInfo component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 30%;
}

.c1 {
  background: #ebebeb;
}

.c2 {
  padding-top: 1rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h3>
          Red tomatoes, raw
        </h3>
        <p>
          Nutritional value per 100 g (3.5 oz)
        </p>
        <table>
          <tbody>
            <tr>
              <th
                class="c1"
              >
                Energy
              </th>
              <th
                class="c1"
              >
                74kJ (18 kcal)
              </th>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Carbohydrates
              </td>
              <td
                class="c2"
              >
                3.9 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                - Sugars
              </td>
              <td
                class="c2"
              >
                2.6 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                - Dietary fiber
              </td>
              <td
                class="c2"
              >
                1.2 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Fat
              </td>
              <td
                class="c2"
              >
                0.2 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Protein
              </td>
              <td
                class="c2"
              >
                0.9 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Water
              </td>
              <td
                class="c2"
              >
                94.5 g
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Vitamin A equiv.
              </td>
              <td
                class="c2"
              >
                42 µg (5%)
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                - lutein and zeaxanthin
              </td>
              <td
                class="c2"
              >
                123 µg
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Vitamin C
              </td>
              <td
                class="c2"
              >
                14 mg (17%)
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Vitamin E
              </td>
              <td
                class="c2"
              >
                0.54 mg (4%)
              </td>
            </tr>
            <tr>
              <td
                class="c2"
              >
                Potassium
              </td>
              <td
                class="c2"
              >
                237 mg (5%)
              </td>
            </tr>
          </tbody>
        </table>
        <p>
          Percentages are relative to US recommendations for adults
        </p>
        <p>
          Source:
           
          <a
            href="http://www.nal.usda.gov/fnic/foodcomp/search/"
          >
             
            USDA Nutrient Database
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ eidNGR"
    >
      <h3>
        Red tomatoes, raw
      </h3>
      <p>
        Nutritional value per 100 g (3.5 oz)
      </p>
      <table>
        <tbody>
          <tr>
            <th
              class="sc-gsTCUz fbHruz"
            >
              Energy
            </th>
            <th
              class="sc-gsTCUz fbHruz"
            >
              74kJ (18 kcal)
            </th>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Carbohydrates
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              3.9 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              - Sugars
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              2.6 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              - Dietary fiber
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              1.2 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Fat
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              0.2 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Protein
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              0.9 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Water
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              94.5 g
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Vitamin A equiv.
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              42 µg (5%)
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              - lutein and zeaxanthin
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              123 µg
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Vitamin C
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              14 mg (17%)
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Vitamin E
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              0.54 mg (4%)
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlfnbm jELKvi"
            >
              Potassium
            </td>
            <td
              class="sc-dlfnbm jELKvi"
            >
              237 mg (5%)
            </td>
          </tr>
        </tbody>
      </table>
      <p>
        Percentages are relative to US recommendations for adults
      </p>
      <p>
        Source:
         
        <a
          href="http://www.nal.usda.gov/fnic/foodcomp/search/"
        >
           
          USDA Nutrient Database
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
